demos/dotnet/dotnet/NugetAction.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+
+namespace dotnet
+{
+    internal static class NugetAction
```

Comment:
These class-level Action concepts seem like they'd be better named Agents or similar. They are not actions themselves, instead being containers/groupings of related actions. This is especially visible in the method of this Type called "DownloadNugetAction" leading to "NugetAction.DownloadNugetAction()". 

I think I'd rather see "NugetAgent.DownloadNuget()". Thoughts?